tests/unit/nupic/support/decorators_test.py

```diff
@@ -22,18 +22,251 @@
 
 """Unit tests for the nupic.support.decorators module.
 
-TODO: add tests for logEntryExit and retry decorators
+TODO: add tests for logEntryExit
 """
 
 
+from mock import patch, call, Mock
 
-from mock import patch, Mock
+from nupic.support.unittesthelpers.testcasebase import unittest
+from nupic.support import decorators
 
 
-from nupic.support.unittesthelpers.testcasebase import unittest
 
+class TestParentException(Exception):
+  pass
 
-from nupic.support import decorators
+class TestChildException(TestParentException):
+  pass
+
+
+class RetryDecoratorTest(unittest.TestCase):
+  """Unit tests specific to retry decorator."""
+
+
+  def mockSleepTime(self, mockTime, mockSleep):
+    """Configures mocks for time.time and time.sleep such that every call
+    to time.sleep(x) increments the return value of time.time() by x.
+    
+    mockTime:     time.time mock
+    mockSleep:    time.sleep mock
+    """
+
+    class _TimeContainer(object):
+      accumulatedTime = 0
+
+    def testTime():
+      return _TimeContainer.accumulatedTime
+
+    def testSleep(duration):
+      _TimeContainer.accumulatedTime += duration
+
+    mockTime.side_effect = testTime
+    mockSleep.side_effect = testSleep
+
+
+  @patch("time.sleep", autospec=True)
+  @patch("time.time", autospec=True)
+  def testRetryNoTimeForRetries(self, mockTime, mockSleep):
+    """Test that when timeoutSec == 0, function is executed exactly once
+    with no retries, and raises an exception on failure.
+    """
+
+    self.mockSleepTime(mockTime, mockSleep)
+
+    _retry = decorators.retry(
+      timeoutSec=0, initialRetryDelaySec=0.2,
+      maxRetryDelaySec=10)
+
+    testFunction = Mock(side_effect=TestParentException("Test exception"),
+                        __name__="testFunction", autospec=True)
+
+    with self.assertRaises(TestParentException):
+      _retry(testFunction)()
+
+    self.assertFalse(mockSleep.called)
+    testFunction.assert_called_once_with()
+
+
+  @patch("time.sleep", autospec=True)
+  @patch("time.time", autospec=True)
+  def testRetryWaitsInitialRetryDelaySec(self, mockTime, mockSleep):
+    """Test that delay times are correct."""
+
+    self.mockSleepTime(mockTime, mockSleep)
+
+    _retry = decorators.retry(
```

Comment:
A leading underscore usually indicates a private object reference in python. I don't think it's appropriately used here (and in the other new test methods here), and it impacts readability of the code. I must have missed this during the review of the same code in the numenta-apps repo.

`retryDecorator` would be a good alternative.